lib/membrane_mp4/muxer/cmaf/track_samples_cache.ex

```diff
@@ -0,0 +1,324 @@
+defmodule Membrane.MP4.Muxer.CMAF.TrackSamplesCache do
+  @moduledoc false
+
+  defstruct state: :collecting,
+            supports_keyframes?: false,
+            collected_duration: 0,
+            collected: [],
+            to_collect_duration: 0,
+            to_collect: []
+
+  @type state_t :: :collecting | :to_collect
+
+  @type t :: %__MODULE__{
+          state: state_t(),
+          supports_keyframes?: boolean(),
+          collected_duration: non_neg_integer(),
+          collected: list(Membrane.Buffer.t()),
+          to_collect: list(Membrane.Buffer.t())
+        }
+
+  @doc """
+  Force pushes the sample into the cache ignoring any timestamps.
+  """
+  @spec force_push(t(), Membrane.Buffer.t()) :: t()
+  def force_push(%__MODULE__{state: :collecting} = cache, sample) do
+    %__MODULE__{
+      cache
+      | collected: [sample | cache.collected],
+        collected_duration: cache.collected_duration + sample.metadata.duration
```

Comment:
The `duration` field in the metadata is calculated and put there:

https://github.com/membraneframework/membrane_mp4_plugin/blob/partial-cmaf-segments-optimization/lib/membrane_mp4/muxer/cmaf.ex#L393

I think we should remove that and operate on timestamps only in this module